rmf_site_editor/src/widgets/view_nav_graphs.rs

```diff
@@ -215,6 +215,7 @@ impl<'w, 's> ViewNavGraphs<'w, 's> {
                         }
                         None => {
                             let future = AsyncComputeTaskPool::get().spawn(async move {
+                                // TODO(luca) change this to use FileDialogServices
```

Comment:
Is this something that needs to be done before we merge, or is it a note for a future improvement?

Reply:
I would suggest leaving it for a future improvement, there is no regression and everything still works it's just that this existing code can probably be cleaned up.
I tried to keep the PR reasonably well scoped (I actually regret bundling in the `FileDialogServices` and workspace saving changes, it would have been a lot slimmer without). It was super low in the priority list because most of these workflows are fairly niche. There is a total of three cases where I added a TODO, in increasing level of "obscureness of the usecase":

* Browsing asset sources for local assets, local assets are not used too much and most of the assets we do are based on Fuel.
* Exporting nav graphs: Can be useful but from what I understand next gen would use directly `.site` files, current gen would probably generate navgraphs through the `Export SDF` feature at build time.
* Exporting lights: This is so niche that I almost wonder if we should get the button out of the UI altogether to declutter it. It exports all the light information into a text file and if I remember correctly was only used in a demo a long time ago.

Reply:
This might be a good thing to open an issue ticket for, maybe with a "good first issue" tag, although that may be subjective since the contributor may have to have some understanding of async.

Reply:
Done with https://github.com/open-rmf/rmf_site/issues/248